jwql/instrument_monitors/common_monitors/bad_pixel_monitor.py

```diff
@@ -0,0 +1,1066 @@
+        dark_jump_files = []
+        dark_fitopt_files = []
+        dark_obstimes = []
+        if dark_raw_files is not None:
+            index = 0
+            badpix_types.extend(badpix_types_from_darks)
+            # In this case we need to run the pipeline on all input files,
+            # even if the rate file is present, because we also need the jump
+            # and fitops files, which are not saved by default
+            for uncal_file, rate_file in zip(dark_raw_files, dark_slope_files):
+                jump_output, rate_output, fitopt_output = pipeline_tools.calwebb_detector1_save_jump(uncal_file, self.data_dir,
+                                                                                                     ramp_fit=True, save_fitopt=True)
+                self.get_metadata(uncal_file)
+                dark_jump_files.append(jump_output)
+                dark_fitopt_files.append(fitopt_output)
+                if self.nints > 1:
+                    #dark_slope_files[index] = rate_output.replace('rate', 'rateints')
+                    dark_slope_files[index] = rate_output.replace('0_ramp_fit', '1_ramp_fit')
+                else:
+                    dark_slope_files[index] = deepcopy(rate_output)
+                dark_obstimes.append(instrument_properties.get_obstime(uncal_file))
+                index += 1
+
+            if len(all_files) == 0:
+                all_files = deepcopy(dark_slope_files)
+            else:
+                all_files = all_files + dark_slope_files
+
+            min_dark_time = min(dark_obstimes)
+            max_dark_time = max(dark_obstimes)
+            mid_dark_time = instrument_properties.mean_time(dark_obstimes)
+
+        # For the dead flux check, filter out any files that have less than
+        # 4 groups
+        dead_flux_files = []
+        if illuminated_raw_files is not None:
+            for illum_file in illuminated_raw_files:
+                ngroup = fits.getheader(illum_file)['NGROUPS']
+                if ngroup >= 4:
+                    dead_flux_files.append(illum_file)
+        if len(dead_flux_files) == 0:
+            dead_flux_files = None
+
+        # Instrument-specific preferences from jwst_reffiles meetings
+        if self.instrument in ['nircam', 'niriss', 'fgs']:
+            dead_search_type = 'sigma_rate'
+        elif self.instrument in ['miri', 'nirspec']:
+            dead_search_type = 'absolute_rate'
+
+        flat_mean_normalization_method = 'smoothed'
+
+        # Call the bad pixel search module from jwst_reffiles. Lots of
+        # other possible parameters. Only specify the non-default params
+        # in order to make things easier to read.
+        output_file = '{}_{}_{}_{}_bpm.fits'.format(self.instrument, self.aperture, self.dark_query_start, self.query_end)
+        output_file = os.path.join(self.output_dir, output_file)
+        bad_pixel_mask.bad_pixels(flat_slope_files=illuminated_slope_files, dead_search_type=dead_search_type,
```

Comment:
Holy parameter space!

Reply:
This is what happens when you get reps from all the instruments in a room every week for months to discuss the *best* way to find bad pixels.....You end up with 17 *best* ways.